File: Reqnroll/Chapter18/18.3-CleanStartPattern/After/WIMP.Specs/Drivers/StubDatabaseDriver.cs
namespace WIMP.Specs.Drivers;

public class StubDatabaseDriver : IDatabaseDriver
{
    public Task EmptyDatabase()
    {
        //nop
        return Task.CompletedTask;
    }

    public Task UpgradeSchemaIfNeeded()
    {
        //nop
        return Task.CompletedTask;
    }
}

File: CucumberJVM/Chapter20/20.3-ParameterizedExecutionPattern/Before/src/test/java/com/wimp/app/specs/drivers/StubDatabaseDriver.java
package com.wimp.app.specs.drivers;

import org.springframework.boot.autoconfigure.condition.ConditionalOnProperty;
import org.springframework.stereotype.Component;

@Component
public class StubDatabaseDriver implements DatabaseDriver {

    @Override
    public void emptyDatabase() {
        // nop
    }
}
